### PR TITLE
Operator modular metrics

### PR DESCRIPTION
**Pull request by pippolo84**
Add a cell to provide global metrics registry and prometheus metrics server for the operator hive.

This makes possible to use `cell.Metric` to export metrics in the operator cells.

Since there are parts of the operator code not yet modularized, the legacy metrics variables and the global registry are kept exported, in order to be directly used by legacy code. This approach is inspired by the agent metrics cell.

Besides, this PR fixes the operator metrics namespace, setting it early at startup, so that all the code shared between other components (e.g: the agent) will report the correct `cilium_operator` namespace when updating the metrics.

Here a query that shows both the correct namespace for all the metrics and also all the metrics registered with `cell.Metric` (like the ones in the `operator/pkg/lbipam` cell) showing up:

```
$ curl localhost:9234/v1/metrics
[
    {
        "name": "cilium_operator_ces_queueing_delay_seconds"
    },
    {
        "name": "cilium_operator_ces_sync_errors_total"
    },
    {
        "name": "cilium_operator_lbipam_conflicting_pools_total"
    },
    {
        "name": "cilium_operator_lbipam_services_matching_total"
    },
    {
        "name": "cilium_operator_lbipam_services_unsatisfied_total"
    },
    {
        "name": "cilium_operator_number_of_ceps_per_ces"
    },
    {
        "name": "cilium_operator_process_cpu_seconds_total",
        "value": 1.03
    },
    {
        "name": "cilium_operator_process_max_fds",
        "value": 1048576
    },
    {
        "name": "cilium_operator_process_open_fds",
        "value": 12
    },
    {
        "name": "cilium_operator_process_resident_memory_bytes",
        "value": 95997952
    },
    {
        "name": "cilium_operator_process_start_time_seconds",
        "value": 1694106767.7
    },
    {
        "name": "cilium_operator_process_virtual_memory_bytes",
        "value": 1351573504
    },
    {
        "name": "cilium_operator_process_virtual_memory_max_bytes",
        "value": 18446744073709552000
    }
]
```

### CODE REVIEW

File: operator/metrics/cell.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+var defaultConfig = Config{
+	// default server address for operator metrics
+	OperatorPrometheusServeAddr: ":9963",
+}
+
+func (def Config) Flags(flags *pflag.FlagSet) {
+	flags.String(OperatorPrometheusServeAddr, def.OperatorPrometheusServeAddr, "Address to serve Prometheus metrics")
+}
+
+type MetricsConfig struct {
```
Comment (joamaki) [inline operator/metrics/cell.go]: This can be removed?

File: operator/metrics/legacy.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	// It is replaced by CiliumEndpointSliceSyncTotal metric.
+	CiliumEndpointSliceSyncErrors = metrics.NoOpCounter
+
+	// CiliumEndpointSliceQueueDelay measures the time spent by CES's in the workqueue. This measures time difference between
+	// CES insert in the workqueue and removal from workqueue.
+	CiliumEndpointSliceQueueDelay = metrics.NoOpHistogram
+)
+
+type legacyMetrics struct {
+	IdentityGCSize                metric.Vec[metric.Gauge]
```
Comment (joamaki) [inline operator/metrics/legacy.go:87]: Let's do a follow-up to remove `legacyMetrics`. Should be pretty doable to do all these after the CES PR lands?

**Review by joamaki — APPROVED** [review]

File: operator/metrics/cell.go (hunk 1 above)
Comment (pippolo84) [reply operator/metrics/cell.go]: Yep, that's just a leftover from a previous iteration :sweat_smile:

File: operator/metrics/legacy.go (hunk 2 above)
Comment (pippolo84) [reply operator/metrics/legacy.go:87]: Definitely. Both the identity GC and the CES controller have been modularized, so we can switch to `cell.Metric` and remove all of this. :+1:

**Comment by pippolo84** [comment]
Rebased to fix conflicts after merging of https://github.com/cilium/cilium/pull/26836

File: operator/metrics/legacy.go [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+			Namespace: metrics.CiliumOperatorNamespace,
+			Name:      "ces_sync_errors_total",
+			Help:      "Number of CES sync errors",
+		}),
+
+		CiliumEndpointSliceSyncTotal: metric.NewCounterVec(metric.CounterOpts{
+			Namespace: metrics.CiliumOperatorNamespace,
+			Name:      "ces_sync_total",
+			Help:      "The number of completed CES syncs by outcome",
+		}, []string{"outcome"}),
```
Comment (derailed) [inline operator/metrics/legacy.go]: Should we use the defined label aka LabelOutcome?

File: operator/metrics/legacy.go [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+		return result, err
+	}
+
+	for _, val := range currentMetrics {
+
+		metricName := val.GetName()
+		metricType := val.GetType()
+
+		for _, metricLabel := range val.Metric {
+			labels := map[string]string{}
```
Comment (derailed) [inline operator/metrics/legacy.go]: nit: prealloc the map? ie make(map[string]string, len(metricLabel.GetLabel()))

**Review by derailed — COMMENTED** [review]
@pippolo84 Nice work Fabio! Just a couple comments...

**Comment by pippolo84** [comment]
> @pippolo84 Nice work Fabio! Just a couple comments...

Thanks for the feedback. Fixed it!

**Review by christarazi — APPROVED** [review]

**Comment by pippolo84** [comment]
Added a commit to remove the stale `enableMetrics` bool in identitygc (no need to check for metrics enabled flag anymore, since we switched to NoOp* types).

**Comment by pippolo84** [comment]
Conformance Ingress failure tracked [here](https://github.com/cilium/cilium/issues/28169), rerunning

**Review by derailed — APPROVED** [review]
@pippolo84 LGTM. Thanks for the updates!

**Review by tommyp1ckles — APPROVED** [review]
IPAM changes look good

**Comment by lmb** [comment]
I resolved a pending conversation to make this mergeable: https://github.com/cilium/cilium/pull/28005/files#r1322543478

**Pull request merged**